.buildkite/scripts/steps/integration_tests.sh

```diff
@@ -19,7 +19,7 @@ else
  OVERRIDE_TEST_AGENT_VERSION=""
 fi
 # PACKAGE
-AGENT_PACKAGE_VERSION="${OVERRIDE_AGENT_PACKAGE_VERSION}" DEV=true EXTERNAL=true SNAPSHOT=true PLATFORMS=linux/amd64,linux/arm64,windows/amd64 PACKAGES=tar.gz,zip mage package
+AGENT_PACKAGE_VERSION="${OVERRIDE_AGENT_PACKAGE_VERSION}" DEV=true EXTERNAL=true SNAPSHOT=true PLATFORMS=linux/amd64,linux/arm64,windows/amd64 mage package
```

Comment:
By removing PACKAGES, I think this also builds all the Docker container variants we don't need. Not wrong, but slower than it could be.

Does adding `deb,rpm` to the packages list or does it cause problems with windows?

Reply:
Is it faster if we just build Windows with a separate invocation of `mage package`? I'm not sure if the underlying cross-compilation is actually parallelized.

Reply:
We could also make the windows target smart enough to ignore the deb and rpm packages if that was in fact the problem.

Reply:
+1

Reply:
Sorry, forgot to add the comment on this.  I tried with `PACKAGES="tar.gz,zip,deb,rpm"` and it failed to build in CI, "compiled mage file failed to run" if I remember correctly.  So I think we have a bug in the mage file, I'm still looking into it.  If I don't find it today or tomorrow I'll open up an issue.

Reply:
Thanks, not opposed to optimizing this separate from merging this PR.

Reply:
So I finally found the timing issue in the magefile for building with the multiple package types defined.  We tried build all the package types at the same time.  So if you have "zip" and the timing is right you can have a compiled magefile that isn't for the GOOS you want.  The "fix" will do the GOOS-ARCH in parallel, so all linux-amd64, then all linux-arm, then windows-amd64.

Reply:
Interestingly the timing vs main is still exactly the same as what it was before (caveat of sample size of 1 still applies):

* 1h52m for a recent build on main https://buildkite.com/elastic/elastic-agent/builds/7645
* 2h11m for the most recent build here https://buildkite.com/elastic/elastic-agent/builds/7624

Blake's suggestion for the future in https://github.com/elastic/elastic-agent/pull/4301#discussion_r1516304403 might be a way to reclaim some of this time but not uploading an unnecessary .tar.gz package. Probably worth putting that in an issue so it isn't lost in this PR.